### PR TITLE
lib: fix bug with huge test output

### PR DESCRIPTION
**Pull request by refack**
Fixes: https://github.com/nodejs/citgm/issues/618

##### Checklist

- [X] `npm test` passes
- [X] contribution guidelines followed [here](https://github.com/nodejs/citgm/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Comment by refack** [comment]
/CC @nodejs/citgm @addaleax PTAL. ATM jobs stall and don't even produce the report.

**Comment by targos** [comment]
Shouldn't we display the last N lines of the output instead of the first? They're more likely to contain useful information about the error

**Comment by refack** [comment]
I think it's arbitrary. In the case of `graseful-fs` the failing test is at the top.
The bottom in actually the concatenated stderr. Which is this case is just noise.

File: lib/reporter/logger.js [hunk 1]
```diff
@@ -11,14 +10,13 @@ function logModule(log, logType, module) {
   }
   if (module.error) {
     log[logType]('error:', module.error.message);
-    log[logType]('error:', module.testOutput);
+    // If the output is too big the log processor explodes.
+    log[logType]('error:', module.testOutput.substring(0, 10000));
```
Comment (MylesBorins) [inline lib/reporter/logger.js]: could we iterate over substrings rather than drop arbitrary content?

Comment (refack) [reply lib/reporter/logger.js]: Ack.
I had the same thought. Definitely worth a try.

Comment (refack) [reply lib/reporter/logger.js]: Done

**Comment by refack** [comment]
> could we iterate over substrings rather than drop arbitrary content?

New code finished in reasonable time:
```
113199ms
```

File: lib/reporter/logger.js [hunk 2]
```diff
@@ -1,24 +1,31 @@
...
-    log[logType]('error:', module.testOutput);
+    // If the output is too big the log formatter explodes.
+    for (let i = 0; i < module.testOutput.length; ) {
+      const rawChunk = module.testOutput.substring(i, i += outChunkSize);
+      // Align chunk to EOL.
+      const lastEOL = rawChunk.lastIndexOf('\n') + 1;
+      i -= (outChunkSize - lastEOL);
+      const chunk = rawChunk.substring(0, lastEOL);
+      log[logType]('error:', chunk);
+    }
```
Comment (BridgeAR) [inline lib/reporter/logger.js]: Would you be so kind and add a more detailed explanation? The current code is somewhat confusing to me.

Comment (refack) [reply lib/reporter/logger.js]: Ack

**Comment by targos** [comment]
Ping @MylesBorins do you still want to review the new code?

**Comment by richardlau** [comment]
Add test(s)?

**Comment by targos** [comment]
Rebased.
CI: https://ci.nodejs.org/view/Node.js-citgm/job/citgm-continuous-integration-pipeline/84/console

**Comment by targos** [comment]
CI is green but it doesn't run on Windows. I'll try to execute it on my computer today. Or maybe @richardlau if you have time?

**Comment by targos** [comment]
Or better: add a Windows machine to https://ci.nodejs.org/job/citgm-continuous-integration/

**Comment by richardlau** [comment]
> 
> 
> Or better: add a Windows machine to https://ci.nodejs.org/job/citgm-continuous-integration/

I'll try to find time to do this later today (I need a few hours sleep first). We could also try adding Windows to the Travis testing.

**Comment by bzoz** [comment]
There is a bug with spawning and PATH (fix in https://github.com/nodejs/citgm/pull/640). With that PR included, this PR works on my local box.

The CITGM tests are broken on Windows. With this PR we will be able to at least run them locally. Anyway - the CITGM is currently completely broken on Windows, this PR at least restores it to a working condition (and looks like nothing will get broken), I would be in favor of fast tracking this.

**Comment by MylesBorins** [comment]
Land away. I'm waiting for a train so can't do so right now. We can revisit
the logic here later if we want to simplify it

**Comment by targos** [comment]
I'm rebasing this

**Comment by bzoz** [comment]
On my local box, tests results:
```
  323 passing (2m)
  27 failing
```

**Comment by targos** [comment]
On mine:
```
  333 passing (3m)
  17 failing
```

**Comment by targos** [comment]
Rebased with https://github.com/nodejs/citgm/pull/619/commits/89c487ac09a3f6cd18e74dfe0ab270f3ebbaa1fb to fix the linter issue

**Comment by targos** [comment]
@bzoz LGTY? Can I land this as-is?

**Comment by bzoz** [comment]
Yes

**Pull request merged**